Migrate to register(storeDescriptor) for plugins data store

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/55374

As part of the React 18 upgrade and WordPress 6.6 compatibility changes, this PR updates the **plugins** store registration pattern in `@woocommerce/data` to use the modern WordPress data store API. 

Key changes include:

1. Migrated from deprecated `registerStore` to `createReduxStore` and `register` pattern
2. Updated store exports to match the modern pattern used across other stores:
    - Added `export { store as pluginsStore }`
    - Maintained `PLUGINS_STORE_NAME` export for backward compatibility
3. Removed deprecated type definitions:
    - Removed custom @wordpress/data module declaration
    - Cleaned up unused type exports and imports
4. Update plugins store import to use `pluginsStore` from `@woocommerce/data`

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This PR doesn't change any functionality, so it's not necessary to test the changes.

For devs:

1. Review the @woocommerce/data changes for the plugins store migration
2. Delete the `noCheck: true` line in `./packages/js/data/tsconfig.json` and running `npx tsc` inside ./packages/js/data
3. Confirm that there are no TS errors in `./src/plugins/**`
4.`window.wp.data.select( wc.data.pluginsStore )` and `window.wp.data.dispatch( wc.data.pluginsStore )` should return selectors and dispatchers


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Migrate to register(storeDescriptor) for plugins data store

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan committed:
Migrate plugins store to use createReduxStore and register

Update plugins store to use the new createReduxStore and register methods from @wordpress/data, similar to recent store migrations. Export the store and update related type definitions.

chihsuan committed:
Update plugins store imports and controls usage

Refactor plugins store to use controls from @wordpress/data directly and update type definitions. Remove deprecated imports and simplify control usage across resolvers and actions.

chihsuan committed:
Update plugins store constant usage across components

Replace PLUGINS_STORE_NAME with pluginsStore in multiple files, ensuring consistent import and usage of the plugins store across various components and task list files. This change follows the recent refactoring of the plugins store and maintains a uniform approach to store imports.

chihsuan committed:
Remove TypeScript type errors in plugins and payments components

Add type-ignore comments for React 18 upgrade and remove unnecessary type error suppression in shipping and payments components

chihsuan committed:
Remove unused imports

chihsuan commented on plugins/woocommerce/client/admin/client/payments-welcome/index.tsx:
```diff
@@ -114,6 +113,7 @@ const ConnectAccountPage = () => {
 			const installAndActivateResponse = await installAndActivatePlugins(
 				[ 'woocommerce-payments' ].concat( pluginsToInstall )
 			);
+			// @ts-expect-error TODO react-18-upgrade: useDispatch does not return the correct type for installAndActivatePlugins.
```
It looks like `useDispatch` does not return the correct type for dispatch functions. It shows a Generator type for return value instead of the actual return type so I ignore the type error for now. I think we need to fix in global scope.  cc @rjchow 

<img width="773" alt="Screenshot 2025-02-13 at 16 11 06" src="https://github.com/user-attachments/assets/e8d01cd6-9bad-408c-870d-b1fab8ead6b7" />

rjchow replied:
Ah ok thanks for catching that, let me have a look at it

adrianduffell reviewed (approved):
Tested well and changes make sense! 🚀

ilyasfoo reviewed (approved):
Nice work here! I also did a quick test of activating plugins via core profiler and deactivating via reactified payment settings, everything works as expected. Thanks, LGTM!

chihsuan committed:
Remove Unused '@ts-expect-error' directive.

chihsuan committed:
Remove TypeScript type error suppression comments in payments welcome page

Clean up unnecessary '@ts-expect-error' directives in the WooCommerce Payments welcome page component, improving type safety and code clarity.

Commit:
Add changefile(s) from automation for the following project(s): @woocommerce/data, @woocommerce/components, woocommerce, woocommerce/client/admin
